web/react/components/team_general_tab.jsx

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) 2015 Spinpunch, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+var SettingItemMin = require('./setting_item_min.jsx');
+var SettingItemMax = require('./setting_item_max.jsx');
+
+var client = require('../utils/client.jsx');
+var utils = require('../utils/utils.jsx');
+
+export default class GeneralTab extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.handleNameSubmit = this.handleNameSubmit.bind(this);
+        this.handleClose = this.handleClose.bind(this);
+        this.onUpdateSection = this.onUpdateSection.bind(this);
+        this.updateName = this.updateName.bind(this);
+
+        this.state = {name: this.props.teamDisplayName, serverError: '', clientError: ''};
+    }
+    handleNameSubmit(e) {
+        e.preventDefault();
+
+        var state = {serverError: '', clientError: ''};
+        var valid = true;
+
+        var name = this.state.name.trim();
+        if (!name) {
+            state.clientError = 'This field is required';
+            valid = false;
+        } else if (name === this.props.teamDisplayName) {
+            state.clientError = 'Please choose a new name for your ' + strings.Team;
+            valid = false;
+        } else {
+            state.clientError = '';
+        }
+
+        this.setState(state);
+
+        if (!valid) {
+            return;
+        }
+
+        var data = {};
+        data.new_name = name;
+
+        client.updateTeamDisplayName(data,
+            function nameChangeSuccess() {
+                this.props.updateSection('');
+                $('#team_settings').modal('hide');
+                window.location.reload();
```

Comment:
Yeah totally agree, was going to make the ticket after this was in.  I looked into trying to fix it with this but getting the sidebar header to update seemed problematic and I didn't want to try and change a bunch of code along with these changes.  Like you said, it's definitely a problem to be fixed later, just will take it's own time to do right.